Revising Send and Sync bounds was a breaking change

`orx-parallel` won't compile anymore `orx-concurrent-ordered-bag` version 2.14.0.

It compiles fine with version 2.13.0.

I get errors like this:

```
error[E0277]: `O` cannot be sent between threads safely
   --> /home/z/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/orx-parallel-2.4.0/src/runner/thread_runner_compute/collect_ordered.rs:31:54
    |
31  |                 Some((idx, value)) => unsafe { o_bag.set_value(offset + idx, map1(value)) },
    |                                                      ^^^^^^^^^ `O` cannot be sent between threads safely
    |
note: required by a bound in `ConcurrentOrderedBag::<T, P>::set_value`
   --> /home/z/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/orx-concurrent-ordered-bag-2.14.0/src/bag.rs:331:12
    |
329 |     pub unsafe fn set_value(&self, idx: usize, value: T)
    |                   --------- required by a bound in this associated function
330 |     where
331 |         T: Send,
    |            ^^^^ required by this bound in `ConcurrentOrderedBag::<T, P>::set_value`
help: consider further restricting type parameter `O` with trait `Send`
    |
19  |     P: IntoConcurrentPinnedVec<O>, O: std::marker::Send
    |                                    ++++++++++++++++++++
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "orx-concurrent-ordered-bag"
-version = "2.14.0"
+version = "3.0.0"
 edition = "2024"
 authors = ["orxfun <orx.ugur.arikan@gmail.com>"]
 description = "An efficient, convenient and lightweight grow-only concurrent data structure allowing high performance and ordered concurrent collection."

